src/Connect.js

```diff
@@ -1,68 +1,441 @@
+    this.did = null
+    this.mnid = null
+    this.address = null
+    this.firstReq = true // Add firstReq?
+    this.doc = null
+
+    // Load any existing state if any
+    if (this.storage)  this.getState()
+    if (!this.keypair) this.keypair = Credentials.createIdentity()
+    if (this.storage)this.setState()
+    // Credential (uport-js) config for verification
+    this.registry = opts.registry || UportLite({ networks: network.config.networkToNetworkSet(this.network) })
+    // TODO can resolver configs not be passed through
+    this.credentials = new Credentials(Object.assign(this.keypair, {registry: this.registry, ethrConfig: opts.ethrConfig, muportConfig: opts.muportConfig }))
+    this.verifyResponse = (token) => {
+      return verifyJWT(token, {audience: this.credentials.did}).then(res => {
+        this.doc = res.doc
+        return this.credentials.processDisclosurePayload(res)
+      })
+    }
+  }
+
+ /**
+  *  Instantiates and returns a web3 styple provider wrapped with uPort functionality.
+  *  For more details see uportSubprovider. uPort overrides eth_coinbase and eth_accounts
+  *  to start a get address flow or to return an already received address. It also
+  *  overrides eth_sendTransaction to start the send transaction flow to pass the
+  *  transaction to the uPort app.
+  *
+  *  @example
+  *  const uportProvider = connect.getProvider()
+  *  const web3 = new Web3(uportProvider)
+  *
+  *  @return          {UportSubprovider}    A web3 style provider wrapped with uPort functionality
+  */
+  getProvider () {
+    // TODO remove defaults, fix import
+    const subProvider = new provider.default({
+      requestAddress: () => {
+        this.requestAddress('addressReqProvider')
+        return this.onResponse('addressReqProvider').then(payload => {
+          this.setDID(payload.res.address)
+          return this.address
+        })
+
+      },
+      sendTransaction: (txObj) => {
+        delete txObj['from']
+        this.sendTransaction(txObj, 'txReqProvider')
+        return this.onResponse('txReqProvider').then(payload => payload.res)
+      },
+      provider: this.provider,
+      networkId: this.network.id
+    })
+    if (this.address) subProvider.setAccount(this.address)
+    return subProvider
+  }
+
+// TODO where to return MNID and where to return address, should this be named differently, will return entire response obj now, not just address
+// TODO requestID? requestAddress? return mnid, address, did in response??
+// TODO add account option, param
+ /**
+  *  Creates a request for only the address/id of the uPort identity.
+  *
+  *  @example
+  *  connect.requestAddress()
+  *
+  *  connect.onResponse('addressReq').then(res => {
+  *    const id = res.res
+  *  })
+  *
+  *  @param    {String}    [id='addressReq']    string to identify request, later used to get response
+  */
+  requestAddress (id='addressReq') {
+    this.credentials.requestDisclosure({callbackUrl: this.genCallback()})
```

Comment:
this should specifically request an `accountType` in this case default it to `keypair`